src/bin/dhcp6/dhcp6_srv.cc

```diff
@@ -3818,12 +3849,88 @@ Dhcpv6Srv::requestedInORO(const Pkt6Ptr& query, const uint16_t code) const {
     return (false);
 }
 
+void Dhcpv6Srv::updateHostKey(AllocEngine::ClientContext6& ctx) {
+    //Default arguments generates a random key
+    AuthKey default_key;
+
+    //check if reservation is present for the client
+    if (ctx.hosts_.empty()) {
+        //create and assign new reservation
+        HostPtr hptr(new Host(ctx.duid_->toText(), "duid",
+                     SubnetID(1), ctx.subnet_->getID(),
+                     IOAddress::IPV4_ZERO_ADDRESS()));
+
+        hptr->setKey(default_key);
+
+        //we are creating new host reservation
+        //ensure to add only if a backend is configured
+        //we dont want to enforce enabling backend 
+        if(!HostMgr::instance().getHostDataSourceList().empty()) { 
+            HostMgr::instance().add(hptr);
+		    ctx.hosts_[ctx.subnet_->getID()] = hptr;
+        }
+		// Now update the context with the new reservation
+    } else {
+        for (auto& host : ctx.hosts_) {
+            if (host.second->getKey().ToText().empty()) {
+                HostPtr hostPtr = boost::const_pointer_cast<Host>(host.second);
+                //multiple host reservation are assigned the same key
+                //as they belong to the same client
+                hostPtr->setKey(default_key);
+            }
```

Comment:
Note it basically does no work because the key is modified only in the memory host object and not in the database.